aptos-move/block-executor/src/view.rs

```diff
@@ -564,82 +652,49 @@ impl<'a, T: Transaction> ResourceState<T> for ParallelState<'a, T> {
                                 Err(e) => {
                                     error!("Couldn't patch value from versioned map: {}", e);
                                     self.captured_reads.borrow_mut().mark_incorrect_use();
-                                    return ReadResult::HaltSpeculativeExecution(
+                                    return Ok(ReadResult::HaltSpeculativeExecution(
                                         "Couldn't patch value from versioned map".to_string(),
-                                    );
+                                    ));
                                 },
                             }
                         }
                     }
 
-                    let data_read = match DataRead::from_value_with_layout(version, value)
-                        .downcast(target_kind)
-                    {
-                        Some(data_read) => data_read,
-                        None => {
-                            error!("Couldn't downcast value from versioned map");
-                            self.captured_reads.borrow_mut().mark_incorrect_use();
-                            return ReadResult::HaltSpeculativeExecution(
-                                "Couldn't downcast value from versioned map".to_string(),
-                            );
-                        },
-                    };
-
-                    if self
-                        .captured_reads
-                        .borrow_mut()
-                        .capture_read(key.clone(), None, data_read.clone())
-                        .is_err()
-                    {
-                        // Inconsistency in recorded reads.
-                        return ReadResult::HaltSpeculativeExecution(
-                            "Inconsistency in reads (must be due to speculation)".to_string(),
-                        );
-                    }
-
-                    return ReadResult::from_data_read(data_read);
+                    return Ok(self.captured_reads.borrow_mut().capture_data_read(
```

Comment:
Seems like this function returns `PartialVMResult` so `capture_data_read` can return partial one as well? Then no need to `?` and add `Ok`, same below